### ISSUE TITLE
Preliminary documentation?

### ISSUE THREAD

**Issue by red8888** [taiidani/terraform-provider-jenkins#45]
This is the top result when I google "jenkins terraform provider". I've seen many efforts to do this and they have all been defeated by the awfulness of Jenkins config lol- but I still get excited when I see someone taking another crack at it.

It would be awesome to add some basic info to the README like the scope of this (looks like your just focusing on the important stuff like jobs/folders/creds which the only thing I'd really want from this). 

Also, maybe something about how it works? Is it making Jenkins REST API calls? How are you diffing the config? It always seemed like that was the tricky part that required you to actually read the xml off the file system or something.

Appreciate this effort!

**Comment by Malet** [comment]
I'm interested in using this too, slightly cautious of using a provider which isn't widespread.

It's using https://github.com/bndr/gojenkins which seems to support most operations, the library uses API calls over HTTP to perform the changes.

**Comment by taiidani** [comment]
Thanks for your interest in the repository! This is a free-time project so I do understand your caution around using it. I'm actively accepting PRs and Issues as well as anyone interested in joining as a maintainer but my turnaround time is measured in days.

I'm a little hesitant to provide too heavy of documentation in the README itself as I see the benefit as small over https://registry.terraform.io/providers/taiidani/jenkins/latest or within the source code at https://pkg.go.dev/github.com/taiidani/terraform-provider-jenkins. That said, the latter in particular could use some work -- I'll look into that!

Lastly, as for scope -- I am accepting PRs for functionality outside of Jenkins Core provided that the documentation for that resource clearly states the plugins required for it to be functional. Example: https://registry.terraform.io/providers/taiidani/jenkins/latest/docs/resources/credential_vault_approle.

tl;dr will these changes fulfill what you are looking for?

- Better go.dev auto-generated documentation
- Clarify the scope of what PRs I accept in the README
- Clarify the current state of maintenance in the README
